component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/backupcode/BackupCodeAPIHandler.java

```diff
@@ -45,128 +41,96 @@
 public class BackupCodeAPIHandler {
 
     /**
-     * Retrieve backup codes for the given username.
+     * Retrieve the count of remaining backup codes for the given username.
      *
      * @param username Username of the user.
-     * @return List of backup code of the user.
+     * @return count of remaining backup code of the user.
      * @throws BackupCodeException If an error occurred while getting backup codes.
      */
-    public static List<String> getBackupCodes(String username) throws BackupCodeException {
+    public static Integer getRemainingBackupCodesCount(String username) throws BackupCodeException {
 
         String tenantAwareUsername;
+        List<String> remainingBackupCodesList = new ArrayList<>();
         try {
             UserRealm userRealm = BackupCodeUtil.getUserRealm(username);
             if (userRealm != null) {
                 tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
                 Map<String, String> userClaimValues = userRealm.getUserStoreManager()
                         .getUserClaimValues(tenantAwareUsername, new String[]{BACKUP_CODES_CLAIM}, null);
-                String encryptedBackupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
-                String backupCodes = null;
-                if (StringUtils.isNotEmpty(encryptedBackupCodes)) {
-                    backupCodes = BackupCodeUtil.decrypt(encryptedBackupCodes);
-                }
-                if (StringUtils.isEmpty(backupCodes)) {
-                    return Collections.emptyList();
-                } else {
-                    return new ArrayList<>(new ArrayList<>(Arrays.asList(backupCodes.split(","))));
+                String backupCodes = userClaimValues.get(BACKUP_CODES_CLAIM);
+                if (backupCodes != null) {
```

Comment:
don't we need to check for blank?

Reply:
addressed with [7d9674b](https://github.com/wso2-extensions/identity-outbound-auth-backup-code/pull/6/commits/7d9674b6bf172b43c30479b0d64fdfa6f8130373)